jupyterhub/handlers/login.py

```diff
@@ -38,8 +40,11 @@ def _render(self, login_error=None, username=None):
     def get(self):
         self.statsd.incr('login.request')
         next_url = self.get_argument('next', '')
+        if next_url.startswith('%s://%s' % (self.request.protocol, self.request.host)):
```

Comment:
Maybe `if next_url.startswith('%s://%s/'` (with a slash on the end) to make sure people can't do e.g. `https://compmodels.tmpnb.org.com`?